### PR TITLE
ENT-4494 split nodeapi tests

### PR DESCRIPTION
**Pull request by blsemo**
As a pre-requisite to harmonize node-api (and as many of its test as possible) between ENT and OS, I have split out any tests that require access to the node implementation from node-api:test to a separate projects (that will not be harmonized/open core).

### CODE REVIEW

File: testing/core-test-utils/build.gradle [hunk 1]
```diff
@@ -9,7 +9,7 @@ description 'Core test types and helpers for testing Corda'
 
 dependencies {
     implementation project(':core')
-    implementation project(':test-common')
+    compile project(':test-common')
```
Comment (mnesbit) [inline testing/core-test-utils/build.gradle]: Hmmm. implementation is the modern gradle to signal non-transitive dependency. Compile does not change this. Did you mean compileOnly, or something? Ask the wizard.

Comment (mnesbit) [reply testing/core-test-utils/build.gradle]: Perhaps you mean api

Comment (blsemo) [reply testing/core-test-utils/build.gradle]: Compile is transitive (in the version that we use). We want this to be transitive, projects that have a test dependency on this also want `test-common` and its transitive dependencies...
I originally turned this to implementation Chris R's request, but that made downstream projects fail test compilation.
When we move to a more modern version of gradle (or javalib), we'll have to go through all of the project and carefully convert compile to implementation or api.

Comment (blsemo) [reply testing/core-test-utils/build.gradle]: Long story short - the wizard said use java-library and api. (api is not available without applying java library)

**Review by mnesbit — APPROVED** [review]
OK

**Pull request merged**